notes.txt

```diff
@@ -0,0 +1,166 @@
+##########################
```

Comment:
Good that you've made this research on your own, and, perhaps, for study project it's ok to put general knowledge in the repo, but for real projects it would be superfluous.